core/user_settings.py

```diff
@@ -59,6 +44,40 @@ def get_list_from_csv(
 
     return mapping
 
+def write_csv_defaults(path: Path, headers: tuple[str, str], default: dict[str, str]=None) -> None:
+    if not path.is_file() and default is not None:
+        with open(path, "w", encoding="utf-8", newline="") as file:
+            writer = csv.writer(file)
+            writer.writerow(headers)
+            for key, value in default.items():
+                writer.writerow([key] if key == value else [value, key])
+
+def track_csv_list(filename: str, headers: tuple[str, str], default: dict[str, str]=None) -> DecoratorT:
+    assert filename.endswith(".csv")
+    path = SETTINGS_DIR / filename
+    write_csv_defaults(path, headers, default)
+
+    def decorator(fn: CallbackT) -> CallbackT:
+        @resource.watch(str(path))
+        def on_update(f):
+            data = read_csv_list(f, headers)
+            fn(data)
+
+    return decorator
+
+# NOTE: this is deprecated, use @track_csv_list instead.
+def get_list_from_csv(
+    filename: str, headers: tuple[str, str], default: dict[str, str] = {}
+):
+    """Retrieves list from CSV"""
+    assert filename.endswith(".csv")
+    path = SETTINGS_DIR / filename
+    write_csv_defaults(path, headers, default)
+
+    # Now read via resource to take advantage of talon's
+    # ability to reload this script for us when the resource changes
+    with resource.open(str(path), "r") as f:
+        return read_csv_list(f, headers)
```

Comment:
We might want to make `get_list_from_csv` raise its own DeprecationWarning (and suppress the `resource.open` one) - users of knausj may be using it directly rather than `resource.open` (I was, for instance), so a more detailed deprecation might be helpful.

Reply:
Can we print out a deprecation warning when this is used?